pkg/util/validation/limits_test.go

```diff
@@ -358,6 +358,67 @@ func TestMaxTotalQueryLengthWithDefault(t *testing.T) {
 	}
 }
 
+func TestMaxPartialQueryLengthWithDefault(t *testing.T) {
+	tenantLimits := map[string]*Limits{
+		"tenant-a": {
+			MaxQueryLength: model.Duration(5 * time.Hour),
+		},
+		"tenant-b": {
+			MaxQueryLength:        model.Duration(2 * time.Hour),
+			MaxPartialQueryLength: model.Duration(1 * time.Hour),
+		},
+	}
+	defaults := Limits{
+		MaxQueryLength:        model.Duration(3 * time.Hour),
+		MaxPartialQueryLength: model.Duration(2 * time.Hour),
+	}
+
+	ov, err := NewOverrides(defaults, NewMockTenantLimits(tenantLimits))
+	require.NoError(t, err)
+
+	for _, tc := range []struct {
+		tenantIDs     []string
```

Comment:
[nit] Since the purpose is testing `MaxPartialQueryLength()`, there's no need to request multiple IDs (we don't even use it in the test case). Below, directly call `MaxPartialQueryLength()` instead of `SmallestPositiveNonZeroDurationPerTenant()`.

```suggestion
		tenantID     string
```